scripts/lib/CIME/case/case.py

```diff
@@ -826,6 +826,11 @@ def configure(self, compset_name, grid_name, machine_name=None,
             logger.info("Machine is {}".format(machine_name))
 
         nodenames = machobj.get_node_names()
+        expect('mpirun' in nodenames,
```

Comment:
We really don't need this since the xmllint will pick up the error in the file before we arrive at this point.  Have you tried intentionally leaving out the mpirun or module_system statements as part of your testing?